odoo/addons/base/models/ir_mail_server.py

```diff
@@ -127,6 +127,24 @@ def _check_smtp_ssl_files(self):
             elif mail_server.smtp_ssl_private_key and not mail_server.smtp_ssl_certificate:
                 raise UserError(_('SSL certificate is missing for %s.', mail_server.name))
 
+    def write(self, vals):
+        if not vals.get('active', True):
+            active_usages = set()
+            for record in self:
+                active_usages.update(record._active_usages_compute())
+            if len(active_usages) > 0:
+                raise UserError(
+                    _('You cannot archive this outgoing mail server because it is still used in the following cases:\n%s.',
+                      '\n'.join(map(lambda u: '- %s' % u, active_usages))))
+        return super(IrMailServer, self).write(vals)
```

Comment:
new modern Odoo guideline I think :) 
```suggestion
        return super().write(vals)
```

Reply:
Thanks, I changed it everywhere I have made a modification like that.